Sources/SmithyHTTPAPI/Headers.swift

```diff
@@ -5,8 +5,28 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-public struct Headers: Sendable {
-    public var headers: [Header] = []
+import class Foundation.NSRecursiveLock
+
+public struct Headers: @unchecked Sendable {
+    private let lock = NSRecursiveLock()
+
+    private var _headers: [Header] = []
+    public var headers: [Header] {
+        get { access { $0 } }
+        set { mutate { $0 = newValue } }
+    }
+
+    private mutating func mutate(_ block: (inout [Header]) -> Void) {
+        lock.lock()
+        defer { lock.unlock() }
+        block(&_headers)
+    }
+
+    private func access<T>(_ block: ([Header]) throws -> T) rethrows -> T {
+        lock.lock()
+        defer { lock.unlock() }
+        return try block(_headers)
+    }
```

Comment:
Above, the `headers` property is protected with exclusive access.  Below, various mutating methods & accessors are modified to use exclusive access.